napari_builtins/_tests/test_io.py

```diff
@@ -68,23 +69,21 @@ def test_guess_zarr_path():
     assert not _guess_zarr_path('no_zarr_suffix/data.png')
 
 
-def test_zarr():
+def test_zarr(tmp_path):
     image = np.random.random((10, 20, 20))
-    with TemporaryDirectory(suffix='.zarr') as fout:
-        z = zarr.open(fout, 'a', shape=image.shape)
-        z[:] = image
-        image_in = magic_imread([fout])
-        # Note: due to lazy loading, the next line needs to happen within
-        # the context manager. Alternatively, we could convert to NumPy here.
-        np.testing.assert_array_equal(image, image_in)
+    data_path = str(tmp_path / 'data.zarr')
+    z = zarr.open(store=data_path, mode='a', shape=image.shape)
+    z[:] = image
+    image_in = magic_imread([data_path])
+    np.testing.assert_array_equal(image, image_in)
 
 
 def test_zarr_nested(tmp_path):
     image = np.random.random((10, 20, 20))
     image_name = 'my_image'
     root_path = tmp_path / 'dataset.zarr'
-    grp = zarr.open(str(root_path), mode='a')
-    grp.create_dataset(image_name, data=image)
+    grp = zarr.open(store=str(root_path), mode='a')
+    grp.create_dataset(image_name, data=image, shape=image.shape)
```

Comment:
```suggestion
    grp.create_array(image_name, data=image, shape=image.shape)
```
See: https://github.com/napari/napari/pull/7215/files#r1810721505